### PR TITLE
333 rename pk parameters

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: R/pk-parameter.R [hunk 1]
```diff
@@ -1,29 +1,70 @@
+#' Standard PK-Parameters types defined in OSPSuite
+#' This is only used to defined how a user defined PK Parameter should be calculated
+#' @include enum.R
+#' @export
+StandardPKParameter <- enum(c(
```
Comment (msevestre) [inline R/pk-parameter.R]: @abdullahhamadeh This enum should not be used as I previously stated. This is just used to know "How" a parameter should be calculated e.g. (cCmax, is max of conc, Auc is Aread under the curve etc..)

File: R/sensitivity-analysis.R [hunk 2]
```diff
@@ -84,10 +74,6 @@ SensitivityAnalysis <- R6::R6Class(
       private$printLine("Variation range", self$variationRange)
       parameterLength <- length(private$.parameterPaths)
       private$printLine("Number of parameters to vary", if (parameterLength > 0) parameterLength else "Will be estimated at run time")
-      private$printLine("Dynamic PK Parameters:")
-      for (dynamicPkParameter in self$allDynamicPKParameters) {
```
Comment (msevestre) [inline R/sensitivity-analysis.R]: The concept of dynamic pk parameter is all gone now

File: R/user-defined-pk-parameter.R [hunk 3]
```diff
@@ -83,26 +55,24 @@ DynamicPKParameter <- R6::R6Class("DynamicPKParameter",
...
+      self$name <- name
+      self$standardPKParameter <- standardPKParameter
     },
     #' @description
     #' Print the object to the console
     #' @param ... Rest arguments.
     print = function(...) {
       private$printClass()
       private$printLine("Name", self$name)
+      private$printLine("DisplayName", self$displayName)
```
Comment (msevestre) [inline R/user-defined-pk-parameter.R]: A PK Parameter (or user defined pk parameter which is a derived class) has a Name, DisplayName, Dimension and displayUnit
Values hould be exported in the table in the display unit

File: R/utilities-pk-parameter.R [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+#' userDefinedPKParameter <- addUserDefinedPKParameter$new(name = "MyAUC", standardPKParameter = StandardPKParameter$AucTend)
+#' userDefinedPKParameter$startTime <- 50
+#' userDefinedPKParameter$endTime <- 80
+#'
+#' # Adds a user defined parameter named MyCMax that will calculat the value of Cmax between the 4th and 5th application
+#' userDefinedPKParameter <- addUserDefinedPKParameter$new(name = "MyCMax", standardPKParameter = StandardPKParameter$Cmax)
+#' userDefinedPKParameter$startApplicationIndex <- 4
+#' userDefinedPKParameter$endApplicationIndex <- 5
+#' @export
+addUserDefinedPKParameter <- function(name, standardPKParameter, displayName = NULL, displayUnit = NULL) {
```
Comment (msevestre) [inline R/utilities-pk-parameter.R]: This is how we create a basic user defined pk parameter

File: R/utilities-pk-parameter.R [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+#'
+#' @param name Name of PK-Parameter to update
+#' @param displayName Optional display name
+#' @param displayUnit Optional display unit. Note that the unit should be defined in unit of the dimension
+#'
+#' @examples
+#'
+#' updatePKParameter("t_max", "MyTmax", "min")
+#' @export
+updatePKParameter <- function(name, displayName = NULL, displayUnit = NULL) {
```
Comment (msevestre) [inline R/utilities-pk-parameter.R]: This is how we update an existing pk parameter (either default or user defined)

File: R/utilities-pk-parameter.R [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+#' @title Returns an instance of a PK-Parameter by name or NULL if the parameter by name is not found
+#'
+#' @param name Name of PK-Parameter to update
+#' @param stopIfNotFound Indicates whether an exception is thrown when the PK-Parameter was not found. Default is \code{TRUE}
+#'
+#' @examples
+#'
+#' pkParameter <- pkParameterByName(name = "t_max")
+#' @export
+pkParameterByName <- function(name, stopIfNotFound = TRUE) {
```
Comment (msevestre) [inline R/utilities-pk-parameter.R]: retruens a pk parameter by name

File: R/utilities-pk-parameter.R [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+
+#' @title Returns the name of all pk parameters defined in the system
+#'
+#' @examples
+#'
+#' pkParameterNames <- allPKParameterNames()
+#' @export
+allPKParameterNames <- function(){
```
Comment (msevestre) [inline R/utilities-pk-parameter.R]: Another usefull function @abdullahhamadeh : It returns all PK PArameter Names defined in the system. 
This is what I meant with maybe we do not need an enum . You could verify whatever inpuyt the uer is giving by checking if it is in this array or not

**Comment by msevestre** [comment]
@abdullahhamadeh and @pchelle Going to merge this so that you can try the latest version
Please create issues for everything that is not ergonomic for you and I'll update

File: R/utilities-pk-parameter.R (hunk 7 above)
Comment (abdullahhamadeh) [reply R/utilities-pk-parameter.R]: Looks good 👍 
So the pk parameters and SA will be calculated for all elements returned by this?

Comment (msevestre) [reply R/utilities-pk-parameter.R]: Correct. This is how it should work. You don't have to worry anymore about PKparameter type (user defined or not)

At least... that's the theory ;-)

Comment (abdullahhamadeh) [reply R/utilities-pk-parameter.R]: I have this list
image

but the pk calculation was done for this subset of the above:
image

**Comment by msevestre** [comment]
Well not all pk parameter apply
Single dosing vs multiple etc.... The list is the superset of all global parameters.

**Comment by msevestre** [comment]
And normalized parameters are not available. It means that the dose per body weight could not be calculated. Is this a real PKSim simulation?

**Pull request merged**